### PR TITLE
bump golang to 1.23.2

### PR DESCRIPTION
**Pull request by iurygregory**
Closes: #52

### CODE REVIEW

**Comment by iurygregory** [comment]
Seems like CI uses 1.22.8 .-.

And I only noticed now the comment in the issue:
NOTE: do not bump go.mod versions for the sake of bumping. They shall be bumped only if language features using Go 1.23 are used in code, or by dependencies force it. 
🤔 So I will think a bit on how to proceed , since this was what I did.

File: go.mod [hunk 1]
```diff
@@ -1,74 +1,75 @@
 module github.com/metal3-io/ironic-standalone-operator
 
-go 1.22.3
+go 1.23.2
```
Comment (tuminoid) [inline go.mod:3]: As mentioned, this is not wanted. Go.mod go directive shall be the lowest accepted by the dependencies or required by used language features in repo. Our production version is defined in Dockerfile, so bumping this makes no sense and make developer life hard.

File: go.mod [hunk 2]
```diff
@@ -1,74 +1,75 @@
...
 	go4.org/netipx v0.0.0-20231129151722-fdeea329fbba
-	golang.org/x/crypto v0.24.0
-	k8s.io/api v0.30.1
-	k8s.io/apimachinery v0.30.1
-	k8s.io/client-go v0.30.1
-	k8s.io/component-base v0.30.1
-	k8s.io/utils v0.0.0-20230726121419-3b25d923346b
-	sigs.k8s.io/controller-runtime v0.18.3
+	golang.org/x/crypto v0.28.0
+	k8s.io/api v0.31.2
```
Comment (tuminoid) [inline go.mod:10]: Also keep Go bump minimum to isolate changes. K8s bump for example is not needed in this PR.

**Review by tuminoid — CHANGES_REQUESTED** [review]
This is more of a Go dependency bump, rather than Golang bump that we're after in #52. 

TBH, you could just remove the go directive bumps here, and retitle it, and its 100% valid k8s/controller-runtime bump PR as is, which is also needed. I suggest we go this route and make another PR for golang bump.

File: go.mod (hunk 1 above)
Comment (iurygregory) [reply go.mod:3]: @tuminoid tks, just to be sure the docker image would be https://hub.docker.com/layers/library/golang/1.23/images/sha256-db933bde327ab6b27f54c29b092ab3bf9276738432ed1bc730b5c9bd98ff33ef?context=explore ?Thanks!

Comment (tuminoid) [reply go.mod:3]: Docker Hub is known to be very confusing about the digest values it shows, so it is easiest to do `docker pull ...` or `podman pull ...` on a amd64 system and use the SHA it gives you. Right now it seems to match what Docker Hub says.

(At time of writing  it is `sha256:ad5c126b5cf501a8caef751a243bb717ec204ab1aa56dc41dc11be089fafcb4f`)

Comment (iurygregory) [reply go.mod:3]: @tuminoid oh ok! Thanks!

**Comment by tuminoid** [comment]
Yup, this fulfills the minimum requirements for a bump.

Did you check the release notes for any changes that could be utilized here, or changes in behavior that needs to be considered?

**Comment by iurygregory** [comment]
> Yup, this fulfills the minimum requirements for a bump.
> 
> Did you check the release notes for any changes that could be utilized here, or changes in behavior that needs to be considered?

I've checked https://tip.golang.org/doc/go1.23 ,  the only thing that caught my eyes was the [Timer changes](https://tip.golang.org/doc/go1.23#timer-changes) , but it shouldn't be a big impact for us.

**Review by tuminoid — COMMENTED** [review]
If there is language specific changes needed or wanted to be utilized, then
/lgtm

**Review by tuminoid — APPROVED** [review]

**Pull request merged**